src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordToString.cs

```diff
@@ -0,0 +1,91 @@
+    /// declaration does not match the expected signature or accessibility, or
+    /// if the explicit declaration doesn't allow overriding it in a derived type and
+    /// the record type is not sealed.
+    /// It is an error if either synthesized or explicitly declared method doesn't
+    /// override `object.ToString()` (for example, due to shadowing in intermediate base types, etc.).
+    /// </summary>
+    internal sealed class SynthesizedRecordToString : SynthesizedRecordObjectMethod
+    {
+        private readonly MethodSymbol _printMethod;
+        public SynthesizedRecordToString(SourceMemberContainerTypeSymbol containingType, MethodSymbol printMethod, int memberOffset, DiagnosticBag diagnostics)
+            : base(containingType, WellKnownMemberNames.ObjectToString, memberOffset, diagnostics)
+        {
+            Debug.Assert(printMethod is object);
+            _printMethod = printMethod;
+        }
+
+        protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymbol> Parameters, bool IsVararg, ImmutableArray<TypeParameterConstraintClause> DeclaredConstraintsForOverrideOrImplementation) MakeParametersAndBindReturnType(DiagnosticBag diagnostics)
+        {
+            var compilation = DeclaringCompilation;
+            var location = ReturnTypeLocation;
+            return (ReturnType: TypeWithAnnotations.Create(Binder.GetSpecialType(compilation, SpecialType.System_String, location, diagnostics)),
+                    Parameters: ImmutableArray<ParameterSymbol>.Empty,
+                    IsVararg: false,
+                    DeclaredConstraintsForOverrideOrImplementation: ImmutableArray<TypeParameterConstraintClause>.Empty);
+        }
+
+        protected override int GetParameterCountFromSyntax() => 0;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var F = new SyntheticBoundNodeFactory(this, this.SyntaxNode, compilationState, diagnostics);
+
+            try
+            {
+                CSharpCompilation compilation = ContainingType.DeclaringCompilation;
+                var stringBuilder = F.WellKnownType(WellKnownType.System_Text_StringBuilder);
+                var stringBuilderCtor = F.WellKnownMethod(WellKnownMember.System_Text_StringBuilder__ctor);
+
+                var builderLocalSymbol = F.SynthesizedLocal(stringBuilder);
+                BoundLocal builderLocal = F.Local(builderLocalSymbol);
+                var block = ArrayBuilder<BoundStatement>.GetInstance();
+                // var builder = new StringBuilder();
+                block.Add(F.Assignment(builderLocal, F.New(stringBuilderCtor)));
+
+                // builder.Append(<name>);
+                block.Add(makeAppendString(F, builderLocal, ContainingType.Name));
+
+                // builder.Append(" { ");
+                block.Add(makeAppendString(F, builderLocal, " { "));
```

Comment:
Could be calculated at compile time: `makeAppendString(F, builderLocal, ContainingType.Name + " { ")`

Reply:
Not as spec'ed. I can add a follow-up question to test plan if you'd like.

---
In reply to: [470174926](https://github.com/dotnet/roslyn/pull/46616#discussion_r470174926) [](ancestors = 470174926)

Reply:
Hum, this one may be debatable. I'll leave as a follow-up question.

---
In reply to: [470290352](https://github.com/dotnet/roslyn/pull/46616#discussion_r470290352) [](ancestors = 470290352,470174926)